packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import { Error } from '@livechat/design-system-icons/react/material';
+import { Trigger, TriggerSize } from './Trigger';
+import { IPickerListItem, PickerList } from './PickerList';
+import { Icon, IconSizeName, IconTypeName } from '../Icon';
+import styles from './Picker.module.scss';
+import cx from 'clsx';
+
+const baseClass = 'picker';
+
+export interface IPickerProps {
+  disabled?: boolean;
+  label?: string;
+  error?: string;
+  options: IPickerListItem[];
+  size?: TriggerSize;
+  placeholder?: string;
+  isRequired?: boolean;
+  onSelect: (selectedItem: IPickerListItem | null) => void;
+}
+
+export const Picker: React.FC<IPickerProps> = ({
+  disabled,
+  error,
+  label,
+  options,
+  size = TriggerSize.Medium,
+  placeholder = 'Select option',
+  isRequired,
+  onSelect,
+}) => {
+  const [isListOpen, setIsListOpen] = React.useState<boolean>(false);
+  const [selectedItem, setSelectedItem] =
+    React.useState<IPickerListItem | null>(null);
+  const [searchPhrase, setSearchPhrase] = React.useState<string | null>(null);
+  const triggerRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const onDocumentClick = (e: MouseEvent) => {
+      if ((e.target as Element).contains(triggerRef.current) && isListOpen) {
+        return setIsListOpen(false);
+      }
+
+      return;
+    };
+
+    document.addEventListener('click', onDocumentClick);
+
+    return () => document.removeEventListener('click', onDocumentClick);
+  }, [isListOpen]);
+
+  const handleOnTriggerClick = () => {
+    if (disabled) {
+      return;
+    }
+
+    return setIsListOpen((prev) => !prev);
+  };
+
+  const handleOnClose = () => {
+    setIsListOpen(false);
+  };
+
+  const handleOnSelect = (item: IPickerListItem) => {
+    setIsListOpen(false);
+    setSelectedItem(item);
+    onSelect(item);
+  };
+
+  const handleOnClearClick = () => {
+    setIsListOpen(false);
+    setSelectedItem(null);
+    onSelect(null);
```

Comment:
What do you think about adding test cases for calling `onSelect` for different scenarios?

I mean:

```
it("should call onSelect with selected item")
it("should call onSelect with null after clearing the selection")
```

?

Reply:
👍 done